### PR TITLE
Add the posibility to install postgresql the official repository.

### PR DESCRIPTION
**Pull request by jochumdev**

### CODE REVIEW

File: pillar.example.sls [hunk 1]
```diff
@@ -1,68 +1,93 @@
 postgresql:
   lookup:
+    repo:
```
Comment (arnisoph) [inline pillar.example.sls:3]: Please use https://github.com/bechtoldt/saltstack-repos-formula

File: pillar.example.sls [hunk 2]
```diff
@@ -1,68 +1,93 @@
...
+      state: managed
+      name: deb http://apt.postgresql.org/pub/repos/apt/ {{ salt['grains.get']('oscodename') }}-pgdg main
+      file: /etc/apt/sources.list.d/postgresql-{{ salt['grains.get']('oscodename') }}.list
+      key_url: https://www.postgresql.org/media/keys/ACCC4CF8.asc
+    client:
+      pkgs:
+        - postgresql-client-9.4
```
Comment (arnisoph) [inline pillar.example.sls:10]: could you create a second pillar example file with this 9.4 example please?

File: postgresql/defaults.yaml [hunk 3]
```diff
@@ -5,10 +5,9 @@ Debian:
       - postgresql
     service:
       name: postgresql
+    cluster:
+      createcluster_bin: pg_createcluster
     config:
-      manage:
```
Comment (arnisoph) [inline postgresql/defaults.yaml:9]: The data in this file can be overriden by pillar. Can we keep the `manage` list?

File: pillar.example.sls (hunk 1 above)
Comment (jochumdev) [inline pillar.example.sls:3]: Yeah, we could but i don't want an external dependency. Do you want?

File: postgresql/server.sls [hunk 4]
```diff
@@ -51,7 +66,6 @@ user_{{ u.name }}:
   {% if 'groups' in u %}
     - groups: {{ u.groups }}
   {% endif %}
-    - encrypt: {{ u.encrypt|default(True) }}
```
Comment (arnisoph) [inline postgresql/server.sls:54]: Do you know something about this parameter?

**Comment by arnisoph** [comment]
Thank you for your efforts! Could you please check the comments?

File: pillar.example.sls (hunk 1 above)
Comment (arnisoph) [inline pillar.example.sls:3]: The thing is we have tens of formulas all doing their own repo management having duplicated code that need to be maintained to keep support for deb/rhel/etc.. As of https://github.com/bechtoldt/saltstack-repos-formula/blob/master/meta.yaml#L9 I plan to extend the repos formula to detect OS automatically. I think it is reasonable to repo mgmt to this formula. IMHO there are some _rare_ cases where external dependencies can be acceptable (user mgmt, repo mgmt). 
What do you think?

File: postgresql/defaults.yaml (hunk 3 above)
Comment (jochumdev) [inline postgresql/defaults.yaml:9]: There was a good reason to move that to pillar, can't remember it tough.... :/

File: postgresql/server.sls (hunk 4 above)
Comment (jochumdev) [inline postgresql/server.sls:54]: It didn't work with 2015.5.2, was the first thing i removed.

File: pillar.example.sls (hunk 2 above)
Comment (jochumdev) [inline pillar.example.sls:10]: Of course.

File: pillar.example.sls (hunk 1 above)
Comment (jochumdev) [inline pillar.example.sls:3]: In this case i'm not a fan of having an external deb on repo mgmt on reason is everything is configurable trough pillar another is saltstack-repos-formula requires salt-modules. So one who just wants this formula needs the other two to.

File: postgresql/defaults.yaml (hunk 3 above)
Comment (jochumdev) [inline postgresql/defaults.yaml:9]: Now i remember :+1: Without it in defaults.yaml you can decide if you want to manage the config files trough salt or not, this is what i needed during development.

Comment (arnisoph) [inline postgresql/defaults.yaml:9]: simply set `manage: []` (empty list) in your pillar. ;)

**Comment by arnisoph** [comment]
ping @pcdummy :)

**Pull request closed without merging**